### PR TITLE
[1138] Add the ability to minimize the content of a site

### PR DESCRIPTION
**Pull request by pcdavid**
- [1138] Add ADR and update CHANGELOG
- [1138] Improve icon border
- [1138] Convert Panel's CSS to Material UI
- [1138] Make Panels handle both sides and instanciate the Sites itself
- [1138] Enforce a minimum width for panels
- [1138] Support expand/collapse of the panels
- [1138] Adapt existing frontend tests

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [x] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [x] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [x] Variables have a proper type
- [x] Functions’ arguments have a proper type
- [x] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [x] `useState<STATE_TYPE>(…)`
- [x] All components have a proper typing for their props
- [x] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [x] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

No backend changes.

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by gcoutable — CHANGES_REQUESTED** [review]
I noticed some strange behavior:
- The panel content may have to disappear below the central panel
chrome_BKIL5aPLO9
- Start to resize from the closed state open the panel to its previous state
chrome_eRgWCsZZuN
- Open a panel more than it can (conflict with the other opened panel) displays a global overflow. From that state, decreasing the same panel size takes some offset to see the panel actually resize
chrome_1oNWVyzqW6

**Comment by pcdavid** [comment]
> I noticed some strange behavior:
> 
>     * The panel content may have to disappear below the central panel
>     * Start to resize from the closed state open the panel to its previous state
>     * Open a panel more than it can (conflict with the other opened panel) displays a global overflow. From that state, decreasing the same panel size takes some offset to see the panel actually resize

Some of these (not sure if all) should be fixed by https://github.com/eclipse-sirius/sirius-components/pull/1174, which also impacts the currently deployed version (hence why it was made in a separate PR).

File: frontend/src/workbench/__tests__/Site.test.tsx [hunk 1]
```diff
@@ -158,3 +166,129 @@ test('should highlight the icon of the selected view', () => {
...
+      />
+    </ThemeProvider>
+  );
+
+  fireEvent.click(screen.getByTestId('viewselector-First'));
+  expect(isIconHighlighted('First')).toBeTruthy();
+  expect(toggleCalled).toBeTruthy();
+});
+
+test("should toggle state when clicking on the selected view's icon (if expanded)", () => {
```
[the remark this answers is not in the document]
Comment (pcdavid) [reply frontend/src/workbench/__tests__/Site.test.tsx:200]: You're right, wrong copy/paste of the previous test's title. I'll push a fix.

[the remark this answers is not in the document]
Comment (pcdavid) [reply frontend/src/workbench/__tests__/Site.test.tsx]: Fixed in a new commit appended to the PR.

File: frontend/src/core/panels/Panels.tsx [hunk 2]
```diff
@@ -10,70 +10,216 @@
...
+    expanded: true,
+    resizablePanelSize: rightPanelInitialSize,
+  });
 
-  const startResize = (event) => {
-    let initialPosition = event.clientX;
-    setState((prevState) => {
-      return { isDragging: true, initialPosition, resizablePanelSize: prevState.resizablePanelSize };
+  const startResizeLeft: React.MouseEventHandler = (event) => {
+    let initialPosition: number = event.clientX;
```
Comment (sbegaudeau) [inline frontend/src/core/panels/Panels.tsx:74]: minor nitpick, this could probably be a `const`

File: frontend/src/core/panels/Panels.tsx [hunk 3]
```diff
@@ -10,70 +10,216 @@
...
+    initialPosition: 0,
+    expanded: true,
+    resizablePanelSize: rightPanelInitialSize,
+  });
 
-  const startResize = (event) => {
-    let initialPosition = event.clientX;
-    setState((prevState) => {
-      return { isDragging: true, initialPosition, resizablePanelSize: prevState.resizablePanelSize };
+  const startResizeLeft: React.MouseEventHandler = (event) => {
```
Comment (sbegaudeau) [inline frontend/src/core/panels/Panels.tsx:73]: `React.MouseEventHandler<HTMLDivElement>` (this way the event will have the proper type for its target)

**Review by sbegaudeau — COMMENTED** [review]
I still need to download the PR to really play with it but I have not seen any issues with the code (only tiny details that I'll fix by myself, don't worry about that)

**Review by sbegaudeau — APPROVED** [review]
I'll squash that and merge it after the build

**Comment by pcdavid** [comment]
Thanks!

**Pull request merged**